### ISSUE TITLE
Unable to restore app

### ISSUE THREAD

**Issue by PookaMustard** [MuntashirAkon/AppManager#293]
**Describe the bug**
Using App Manager, I made a backup of an old version of Jetpack Joyride. The backup was created successfully, but attempts to restore it have been met with failure. This is in root mode.

**To Reproduce**
Steps to reproduce the behaviour:
1. Backup the app as a multi-backup
2. Uninstall app
3. Restore app (APK and data).

**Expected behavior**
Jetpack Joyride installed with all the backed up data reinstated.

**Screenshots**
![Screenshot_20210222-093335](https://user-images.githubusercontent.com/19552979/108677042-ca9a9200-74f1-11eb-9789-b1c7a6270d27.jpg)

**Logcat logs**
[logcat_02-22-2021_09-31-12.txt](https://github.com/MuntashirAkon/AppManager/files/6020214/logcat_02-22-2021_09-31-12.txt)

**Device info**
 - Device: Huawei Mate 10 lite
 - OS Version: Android 8.0
 - App Manager Version: v2.5.24-DEBUG#922 plus previous debugs
 - Mode: root

**Additional context**
Inspection of the logcat output show that attempting to extract the archive containing the app has failed, probably `caused by: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died`. GZip compression was used. This error persisted even installing several updates to the Debug build as well as uninstalling and reinstalling AM, plus two or three system restarts and several attempts at force closing AM.

**Comment by MuntashirAkon** [comment]
Need a logcat output without any filters. Send me via PM.

**Comment by PookaMustard** [comment]
I'm not sure if there's a way to send PMs over GitHub, but I sent the unfiltered logcat over Telegram.

**Comment by PookaMustard** [comment]
Apparently not even data-only restores work. It produces the same logcat outputs as trying to restore APK + data.

**Comment by MuntashirAkon** [comment]
Yes, that wouldn't work either. The issue doesn't seem to occur on my device. So, I guess it's a device/OS specific issue.

**Comment by PookaMustard** [comment]
That's awkward.

I'll give it a try on my Android 7 tablet and see what happens.

**Comment by PookaMustard** [comment]
I tried it on my Android 7 tablet today. Here's a new logcat.

[logcat_02-27-2021_05-06-26.txt](https://github.com/MuntashirAkon/AppManager/files/6053740/logcat_02-27-2021_05-06-26.txt)

Somehow I get a different error than before now. I'll DM you the full logcat later if needed.

**Comment by PookaMustard** [comment]
v2.5.24-DEBUG#954 still suffers from the issue.

I'll probably take a look at the code to see if there's anything I can help with, but would probably need something for more advanced debugging than logcat.

**Comment by MuntashirAkon** [comment]
I actually haven't looked into the issue yet. I've been very busy of late.

**Comment by MuntashirAkon** [comment]
> Somehow I get a different error than before now. I'll DM you the full logcat later if needed.

Wrong mode of operations?

**Comment by PookaMustard** [comment]
I left it on auto, and just now I switched to root operations. Still no dice.

**Comment by MuntashirAkon** [comment]
> I left it on auto, and just now I switched to root operations. Still no dice.

Has root been detected?

**Comment by PookaMustard** [comment]
> > I left it on auto, and just now I switched to root operations. Still no dice.
> 
> Has root been detected?

Yes. I granted it through Magisk, and in the about device section of the app, Root is true.

**Comment by MuntashirAkon** [comment]
> Yes. I granted it through Magisk, and in the about device section of the app, Root is true.

Ah, so root functions are working as expected except backup/restore?

**Comment by PookaMustard** [comment]
> > Yes. I granted it through Magisk, and in the about device section of the app, Root is true.
> 
> Ah, so root functions are working as expected except backup/restore?

I think so? I was able to uninstall an app without the system itself prompting me to confirm it.

**Comment by MuntashirAkon** [comment]
> I think so? I was able to uninstall an app without the system itself prompting me to confirm it.

Then the logs seem unrelated to backup/restore.

**Comment by PookaMustard** [comment]
Issue was resolved in a previous commit (no idea which).

**Comment by MuntashirAkon** [comment]
Related commits 42f1174d and 82cd9f29.

**Comment by PookaMustard** [comment]
I had the issue happen again, although it seems to be down to calling for root.

I don't know, I tried replicating it many times but couldn't get it again. At one point, I uninstalled the app, but didn't get prompted for root. Then I went to restore an app and got a failure, then switched the mode of operation to root and then restored - again, failure. When I tried to backup an app, only then did I get prompted for root. And since then, everything worked smoothly, and I couldn't replicate the issue again.

This is with debug 1006.
